### PR TITLE
Add `overline` style

### PR DESCRIPTION
**Pull request by Yash-Singh1**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -139,7 +144,7 @@ declare namespace ansiStyles {
 
 		@param hex - A hexadecimal string containing RGB data.
 		*/
-		hexToRgb(hex: string): [red: number, green: number, blue: number];
+		hexToRgb(hex: string): [number, number, number];
```
Comment (sindresorhus) [inline index.d.ts:147]: Don't do unrelated changes

File: index.d.ts [hunk 2]
```diff
@@ -48,6 +48,11 @@ declare namespace ansiStyles {
 		*/
 		readonly underline: CSPair;
 
+		/**
+		Make text overline. (Not widely supported)
```
Comment (sindresorhus) [inline index.d.ts:52]: This should list the supporting terminals.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: index.d.ts (hunk 1 above)
Comment (Yash-Singh1) [reply index.d.ts:147]: The tests seemed to fail without this change, but I will revert it.

File: index.d.ts (hunk 2 above)
Comment (Yash-Singh1) [reply index.d.ts]: e20eb46

File: index.d.ts [hunk 3]
```diff
@@ -48,6 +48,11 @@ declare namespace ansiStyles {
 		*/
 		readonly underline: CSPair;
 
+		/**
+		Make text overline. Supported on VTE-based terminals, the GNOME terminal, mintty, and Git Bash.
```
Comment (sindresorhus) [inline index.d.ts:52]: The readme and index.d.ts should be in sync.

Comment (Yash-Singh1) [reply index.d.ts:52]: bcf6fe8
